llmfoundry/data/finetuning/tasks.py

```diff
@@ -339,14 +339,19 @@ def dataset_mapper(example: Dict):
                 example = preprocessing_fn(example)
             return _tokenize_formatted_example(example, tokenizer)
 
+        detected_cpu_count = os.cpu_count() or 1
+
         columns_to_remove = list(dataset[0].keys())
         tokenized_dataset = dataset.map(
             dataset_mapper,
             batched=False,
             remove_columns=columns_to_remove,
+            num_proc=max(1, detected_cpu_count - 4),
```

Comment:
why minus 4?

Reply:
totally arbitrary choice to not use all the cpus 🤷‍♂️